src/js/mods/modloader.js

```diff
@@ -269,6 +290,73 @@ export class ModLoader {
         this.modLoadQueue = [];
         this.initialized = true;
     }
+
+    /**
+     *
+     * @param {string} modID
+     */
+    disableMod(modID) {
+        let mod = this.mods.find(mod => mod.metadata.id === modID);
+        if (mod == undefined) {
+            throw new Error("Mod not found");
+        }
+        const index = this.modEnableQueue.indexOf({
+            meta: mod.metadata,
+            path: mod.path,
+        });
+        if (index > -1) {
+            this.modEnableQueue.splice(index, 1);
+        } else if (!mod.disabled) {
+            this.modDisableQueue.push({
+                meta: mod.metadata,
+                path: mod.path,
+            });
+        }
+    }
+
+    /**
+     *
+     * @param {string} modID
+     */
+    enableMod(modID) {
+        let mod = this.mods.find(mod => mod.metadata.id === modID);
+        if (mod == undefined) {
+            throw new Error("Mod not found");
+        }
+        const index = this.modDisableQueue.indexOf({
+            meta: mod.metadata,
+            path: mod.path,
+        });
+
+        if (index > -1) {
+            this.modDisableQueue.splice(index, 1);
+        } else if (mod.disabled) {
+            this.modEnableQueue.push({
+                meta: mod.metadata,
+                path: mod.path,
+            });
+        }
+    }
+
+    async applyModEnableDisable() {
+        const storage = G_IS_STANDALONE
+            ? new StorageImplElectron(this.app)
+            : new StorageImplBrowserIndexedDB(this.app);
+        await storage.initialize();
+        for (let modObj of this.modDisableQueue) {
+            if (this.modEnableQueue.some(value => value.meta == modObj.meta && value.path == modObj.path))
+                continue;
+            let text = await storage.readFileAsync(modObj.path);
+            await storage.writeFileAsync(`${modObj.path}.disabled`, text);
+            await storage.deleteFileAsync(modObj.path);
+        }
+        for (let modObj of this.modEnableQueue) {
+            let text = await storage.readFileAsync(modObj.path);
+            await storage.writeFileAsync(modObj.path.replace(".disabled", ""), text);
+            await storage.deleteFileAsync(modObj.path);
+        }
```

Comment:
manipulating filenames is extremely hacky and can easily lead to hard-to-diagnose issues if discrepancies arise with how the mod folder works (again just have a separate file that keeps track of this stuff)